### ISSUE TITLE
Release 4.7.1

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1386]
Can be labelled **ready** when all other issues on this milestone are closed.

- [x] Address all coding and style violations detected during a *Release* build (send a regular pull request which must be merged before continuing)
- [x] ensure completeness of draft [release notes](https://github.com/FakeItEasy/FakeItEasy/releases), including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
- [x] run `tools\prepare_release.cmd x.y.z` to:
    - change "vNext" names to appropriate x.y.z version for the next GitHub Release, milestone, and release checklist (this issue)
    - create a new draft GitHub Release, milestone, and release checklist
    - create release/x.y.z branch on upstream, branching from develop
    - submit a pull request to merge the release/x.y.z branch to master
- [x] merge the pull request, _which may be done by the PR creator, unlike a regular pull request_
- [x] tag master with the new version number and push to upstream. This will cause the build server to
    - push the nupkgs to [nuget.org](https://www.nuget.org/profiles/FakeItEasy) and
    - attach all nupkgs from the tagged build to the GitHub Release.
- [x] wait for NuGet package validation
- [x] de-list pre-release NuGet packages if present
- [x] publish the GitHub release
- [x] tweet, mentioning contributors, and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- [ ] close this issue
- [ ] close the [milestone](https://github.com/FakeItEasy/FakeItEasy/milestones?state=open) for this release

**Comment by blairconrad** [comment]
I'd like to release as soon as we have #1388/#1390/#1391 fixed, if there's no objection. The current state of `in`-parameter handling is embarrassing.

**Comment by blairconrad** [comment]
Also, I think we'll have a "special thanks from" on this one, and unless there are reasons not to, I'm inclined to release as soon as we get one of those. It gives the contributor nice early feedback before they forget about their contribution. Also, less chance they have to share the laurels with someone else.

**Comment by thomaslevesque** [comment]
> I'd like to release as soon as we have #1388/#1390/#1391 fixed, if there's no objection. The current state of in-parameter handling is embarrassing.

Agreed

> Also, I think we'll have a "special thanks from" on this one, and unless there are reasons not to, I'm inclined to release as soon as we get one of those. It gives the contributor nice early feedback before they forget about their contribution. Also, less chance they have to share the laurels with someone else.

Works for me

**Comment by blairconrad** [comment]
Proposed tweet:

```
Get FakeItEasy 4.8.0!
New:
  Rejects attempts to assign values to read-only in parameters for all fakes on all platforms

With thanks to @zvirja

https://github.com/FakeItEasy/FakeItEasy/releases/4.8.0
```

**Comment by blairconrad** [comment]
bug/enhancement is hard. I was thinking that really complaining about the attempt to modify the value was a service above and beyond just preventing the modification. But silently suppressing the change is kind of a jerk move. Okay!

**Comment by blairconrad** [comment]
Oh. I amended in place. Shoulda just added.

**Comment by blairconrad** [comment]
https://twitter.com/fakeiteasyfx/status/1015177377520013313

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.7.1](https://github.com/FakeItEasy/FakeItEasy/releases/4.7.1).
